golang/connection_test.go

```diff
@@ -168,5 +168,8 @@ func withTestChannel(t *testing.T, f func(ch *Channel)) {
 
 	go ch.ListenAndHandle()
 
-	f(ch)
+	func() {
```

Comment:
You can get rid of the wrapping func - since you are calling it immediately, it isn't doing anything for you.